pandas/core/generic.py

```diff
@@ -4412,6 +4412,29 @@ def _clip_with_scalar(self, lower, upper, inplace=False):
         else:
             return result
 
+    def _clip_with_one_bound(self, threshold, method, axis, inplace):
+
+        inplace = validate_bool_kwarg(inplace, 'inplace')
+
+        if np.any(isnull(threshold)):
+            raise ValueError("Cannot use an NA value as a clip threshold")
+
+        # method is self.le for upper bound and self.ge for lower bound
+        if is_scalar(threshold) and is_number(threshold):
+            if method.__name__ == 'le':
+                return self._clip_with_scalar(None, threshold, inplace=inplace)
+            return self._clip_with_scalar(threshold, None, inplace=inplace)
+
+        subset = method(threshold, axis=axis) | isnull(self)
+
+        # GH #15390
+        if (not isinstance(threshold, ABCSeries)) and is_list_like(threshold):
+            if isinstance(self, ABCSeries) or axis == 0:
+                threshold = pd.Series(threshold, index=self.index)
+            elif axis == 1:
```

Comment:
Thanks for mentioning `_get_axis(axis)`! 

After avoiding to check axis value, I find another untested edge case. When a frame passed to `threshold` and 0 to `axis`, the `threshold` will be transformed to series and lead to error. I have parameterized an existing test to cover this case.

As result, it will require more checks, such as if `self` and `threshold` are series or frame and if `axis` is `None`. Thus, to avoid too many ifs, I used  ` _align_method_FRAME` to do this transform. Let me know this alright.